google-cloud-spanner/lib/google/cloud/spanner/instance.rb

```diff
@@ -188,8 +206,55 @@ def labels= labels
           )
         end
 
+        ##
+        # Update changes.
+        #  `display_name`, `labels`, `nodes`, `processing_units` can be
+        #  updated. `processing_units` and `nodes` can be used interchangeably
+        #  to update.
+        #
+        # @return [Instance::Job] The job representing the long-running,
+        #   asynchronous processing of an instance update operation.
+        # @raise [ArgumentError] if both processing_units or nodes are specified.
+        #
+        # @example
+        #   require "google/cloud/spanner"
+        #
+        #   spanner = Google::Cloud::Spanner.new
+        #
+        #   instance = spanner.instance "my-instance"
+        #   instance.display_name = "prod-instance"
+        #   instance.labels = { env: "prod", app: "api" }
+        #   instance.nodes = 2
+        #   # OR
+        #   # instance.processing_units = 500
+        #
+        #   job = instance.save
+        #
+        #   job.done? #=> false
+        #   job.reload! # API call
+        #   job.done? #=> true
+        #
+        #   if job.error?
+        #     status = job.error
+        #   else
+        #     instance = job.instance
+        #   end
+        #
         def save
```

Comment:
Can we provide a `field_mask` to `save` method? So we can avoid the change of line 246-255.

Reply:
This might not be a good idea. I was just trying to minimize the changes.

Reply:
Yes we can provide optional argument for `field_mask`, but for usability wise it looks like it is not good. User need to first set the field and then need to pass field mask.

Reply:
As per new updated changes by default, it will updates `[display_name node_count labels]`.  If the user has updated fields then it will check for changes for respective fields and build field mask array,

Reply:
An issue with the current implementation is that users can't change the value twice. Let's say `processing_units` = 500 at the beginning, then we update it to `800`. After that, we change it again to `500` (this will fail because the value is equal to the original value `500`). You need to update `original_values` when the new values are saved.

Reply:
Updated original value after update request. One catch here is if a job fails then it will not revert back to original values.

Reply:
Is there any way that we don't store `original_values`? Another problem I notice is that if no thing changes, then `field_mask` will be empty. When we pass an empty array of `field_mask` to `update_instance`, which will use the default config `%w[display_name node_count labels]`.

Reply:
@hengfengli Is there any way to convert nodes to processing units? If there is a formula then calculate processing units and update only `processing_units`.

Reply:
> @hengfengli Is there any way to convert nodes to processing units? If there is a formula then we calculate processing units and update only `processing_units`.

No, I don't think it can work in this way.

Reply:
Actually, I am wrong. 1 node == 1000 processing_units. If we create 5000 processing unites, then it will be equal to 5 nodes.

Reply:
But I am a bit hesitate to hardcode this value into ruby client lib. I hope we don't do this.

Reply:
Can we remove `original_values`? and if `processing_units` is not empty, we use `processing_units`. If `processing_units` is empty, then we use `node_count`. What do you think?

Reply:
Processing units less than 1000 then api retuning 0 node count in response. If it is 1000 then returning node count is 1. 
If processing unit set to >1000 then node count is always present.

Reply:
Yeah, that's correct. But I don't want to write any thresholds in the code. So probably we just keep with the current `original_values` implementation. Instead of calling it `original_values`, may consider `current_values`.

Reply:
ok

Reply:
renamed original values to `current_values`